### PR TITLE
add expand

### PR DESCRIPTION
**Pull request by dding3**
Seprate expand part from https://github.com/intel-analytics/analytics-zoo/pull/966/

### CODE REVIEW

**Comment by dding3** [comment]
CC @zhichao-li

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalExpandSpec.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.keras.layers.internal.InternalExpand
+import com.intel.analytics.zoo.pipeline.api.keras.serializer.ModuleSerializationTest
+
+
+class InternalExpandSpec extends KerasBaseSpec {
+  "InternalExpand" should "generate correct output" in {
+    val tgtSizes = Array(5, 4, 3)
+    val layer = InternalExpand[Float](tgtSizes)
+    val input = Tensor[Float](5, 4, 1).rand()
```
Comment (zhichao-li) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalExpandSpec.scala]: - add a unitest for `includeBatch=true`? i.e `tgtSizes = Array(5, 4, 3), input = Tensor[Float](1, 4, 3).rand()`
- Not sure if it's good that the dimension of `tgtSizes` doesn't respect the value of `includedBatch`, any comments?

Comment (dding3) [reply zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalExpandSpec.scala]: Thanks for your suggestions. Added the unit test with `includeBatch=true`
`includeBatch`  is used to config if we need expand the batch dim, as when we build the model, the batch dim is -1. Rename it to `expandBatchDim` to bring less confusing.

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalExpandSpec.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.keras.layers.internal.InternalExpand
+import com.intel.analytics.zoo.pipeline.api.keras.serializer.ModuleSerializationTest
+
+
+class InternalExpandSpec extends KerasBaseSpec {
+  "InternalExpand" should "generate correct output" in {
+    val tgtSizes = Array(5, 4, 3)
```
Comment (zhichao-li) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalExpandSpec.scala]: What if user input `val tgtSizes = Array(50000, 4, 3) val input = Tensor[Float](5, 4, 1).rand()`  when `includedBatch=false` ? Wondering should we change the tgtSizes to Array(4, 3) in this case ?

**Comment by dding3** [comment]
Synced with @zhichao-li offline and we agreed to use -1 to mask dim which we don't want to expand.

**Comment by zhichao-li** [comment]
LGTM

**Comment by dding3** [comment]
Jekins pass. http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1593/

Thanks for the review.
